header.go

```diff
@@ -471,13 +480,21 @@ func (h *ResponseHeader) AddTrailerBytes(trailer []byte) {
 		for len(key) > 0 && key[len(key)-1] == ' ' {
 			key = key[:len(key)-1]
 		}
+		// Forbidden by RFC 7230, section 4.1.2
 		if isBadTrailer(key) {
+			badTrailers = append(badTrailers, key...)
+			badTrailers = append(badTrailers, strCommaSpace...)
 			continue
 		}
 		h.bufKV.key = append(h.bufKV.key[:0], key...)
 		normalizeHeaderKey(h.bufKV.key, h.disableNormalizing)
 		h.trailer = appendArgBytes(h.trailer, h.bufKV.key, nil, argsNoValue)
 	}
+
+	if len(badTrailers) > 0 {
+		return fmt.Errorf("forbidden trailers: %s", badTrailers[:len(badTrailers)-len(strCommaSpace)])
```

Comment:
I think it's better to define a ʼErrBadTrailerʼ and return that so we don't have any allocations here either. Users can figure out themselves which trailer was bad.